### PR TITLE
Debugging documentation

### LINKED ISSUE #184: Add documentation on how to debug jvm code
It's not obvious how one can debug his code.
We need to add documentation and examples how this works. The commandline docs is not enough for this.
Related to #172

### PR DESCRIPTION
**Pull request by chippmann**
Resolves #184 .

Adds documentation on how to debug jvm code and also changes the supplied jvm args to suspend the execution until a remote debugger is attached.

### CODE REVIEW

**Review by CedNaru — APPROVED** [review]

**Review by piiertho — APPROVED** [review]

File: docs/src/doc/user-guide/debugging.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## Prerequisites
+If you want to debug your code and you're already using an embedded jre, make sure you've included the following modules:
```
Comment (raniejade) [inline docs/src/doc/user-guide/debugging.md:2]:
```suggestion
In order to debug your code using an embedded JRE, make sure to include the following modules:
```

File: docs/src/doc/user-guide/debugging.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+- `jdk.jdwp.agent`
+
+!!! info
+    Example: `jlink --add-modules java.base,java.logging,jdk.jdwp.agent --output jre`
+
+If you still rely on your installed jdk in `JAVA_HOME` you can just follow on.
```
Comment (raniejade) [inline docs/src/doc/user-guide/debugging.md:9]: nit: probably we don't need to mention this.

File: src/gd_kotlin.cpp [hunk 3]
```diff
@@ -211,7 +211,7 @@ void GDKotlin::init() {
         }
 
         String debug_command{
-                "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=" + jvm_debug_address + ":" +
+                "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=" + jvm_debug_address + ":" +
```
Comment (raniejade) [inline src/gd_kotlin.cpp:214]: I think suspend={y|n} should be another CLI flag. It is sometimes useful to attach to the debugger later on.

**Review by raniejade — CHANGES_REQUESTED** [review]

File: src/gd_kotlin.cpp (hunk 3 above)
Comment (chippmann) [reply src/gd_kotlin.cpp]: Made it configurable now, but defaulted it to `y`

**Review by raniejade — APPROVED** [review]

**Review by CedNaru — APPROVED** [review]

File: src/gd_kotlin.cpp [hunk 4]
```diff
@@ -162,6 +162,7 @@ void GDKotlin::init() {
     String jvm_jmx_port;
     bool is_gc_force_mode{false};
     bool is_gc_activated{true};
+    bool is_waiting_for_debugger{true};
```
Comment (piiertho) [inline src/gd_kotlin.cpp:165]: Why is it true by default ?

Comment (piiertho) [reply src/gd_kotlin.cpp:165]: Did not read former conv with ranie, don't mind last comment

**Pull request merged**